include/baresip.h

```diff
@@ -870,6 +870,17 @@ enum ua_event {
 	UA_EVENT_MAX,
 };
 
+
+struct event_param {
+	const char *txt;
+	union {
+		struct ua *ua;
+		struct call *call;
+		struct sip_msg *msg;
+	} u;
```

Comment:
Maybe adding a direct `type` field for the union would be better, a wrong assumption can easily lead to segfaults or strange bugs.